### PR TITLE
Replace FindBugs with SpotBugs maven plugin

### LINKED ISSUE #10406: Swap FindBugs plugin with SpotBugs
**Summary**

The original codehaus FindBugs plugin is deprecated and no longer under active maintenance - https://gleclaire.github.io/findbugs-maven-plugin/

The successor to findbugs is SpotBugs - https://spotbugs.github.io/ It contains more checks for poor patterns throughout the codebase and is actively maintained.

We should use the newer plugin. I made a small attempt at switching, and spotbugs has found nearly 200 issues which findbugs didn't. We should switch to this ASAP.

**Urgency**

Help reduce common bugs across our code base.

### PR DESCRIPTION
**Pull request by ZacBlanco**
Findbugs is deprecated. This PR switches the original FindBugs plugin
to the newer and better-maintained SpotBugs. Additionally, the original
list of bug exclusions has been revised, and a number of bugs fixed
throughout the codebase.

The original build flag `-Dfindbugs.skip` will still work after this change.

Fixes #10406

### CODE REVIEW

**Comment by yuzhu** [comment]
```
Tests in error: 
alluxio.client.cli.fs.JobServiceFaultToleranceShellTest.distributedMv(alluxio.client.cli.fs.JobServiceFaultToleranceShellTest)
  Run 1: JobServiceFaultToleranceShellTest.before:52 » IO java.lang.RuntimeException: a...
  Run 2: JobServiceFaultToleranceShellTest.before:52 » IO java.lang.RuntimeException: a...

Flaked tests: 
alluxio.client.cli.fs.command.PersistCommandTest.persistOnRenameDirectory(alluxio.client.cli.fs.command.PersistCommandTest)
  Run 1: PersistCommandTest.persistOnRenameDirectory:105->AbstractFileSystemShellTest.checkFilePersisted:306
  Run 2: PASS

alluxio.server.ft.journal.MultiMasterJournalTest.testTtl(alluxio.server.ft.journal.MultiMasterJournalTest)
  Run 1: MultiMasterJournalTest.before:53 » IO java.lang.RuntimeException: alluxio.exce...
  Run 2: PASS 
```

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by ZacBlanco** [comment]
I think before merging this PR we should re-evaluate the exclusions/inclusions within the `build/findbugs/findbugs-exclude.xml`

**Comment by yuzhu** [comment]
@ZacBlanco could we remove all the bug code and see what the output looks like? so we can decide which bug code to keep?

**Comment by wyykwan** [comment]
jenkins, test this pleaseeeee

**Comment by wyykwan** [comment]
jenkins, test this please

**Comment by ZacBlanco** [comment]
@gpang @yuzhu FYI

So I found while running this PR that we have a ton of locations that findbugs complains about with the error code [`REC_CATCH_EXCEPTION`](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#rec-exception-is-caught-when-exception-is-not-thrown-rec-catch-exception)

The short explanation is that there's are many places throughout our codebase where we do something like

```java
try {
  // Code that might throw IOException, NPE, NumberFormatException...
} catch (Exception e) { // This exception literally catches everything
 // Generic exception handler
}
```

Spotbugs' explanation of this is essentially there is rarely a time when it is desirable to catch/handle all exceptions, regardless of class. Most often you'll at least want to handle `RuntimeException` separately from `Exception` because they are different classes of errors.

Across the project we do a pretty poor job (according to Spotbugs) of exception handling as there are probably 30+ locations where there is simply `catch (Exception e)`. I tried to fix a lot of them, but I'm worried that I don't have enough context on how this could affect the reliability of the program, such as introducing crashes due to not catching the proper exception class. Nor do I properly know how the different classes of exceptions should be handled in all the different cases I encountered.

What I'm going to do is revert any changes related to `REC_CATCH_EXCEPTION` and file a ticket that can be addressed in a separate PR to deal with our poor exception handling.

**Comment by gpang** [comment]
@ZacBlanco That sounds like a good plan. Let's get the simpler fixes/changes in first, and address the bigger issues separately. Thanks!

**Comment by ZacBlanco** [comment]
Created https://github.com/Alluxio/alluxio/issues/10459 to track the debt for exception handling.

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java [hunk 1]
```diff
@@ -137,7 +137,10 @@ private UnderFileSystem getOrAdd(AlluxioURI ufsUri, UnderFileSystemConfiguration
         fs = new ManagedBlockingUfsForwarder(fs);
       }
 
-      mUnderFileSystemMap.putIfAbsent(key, fs);
+      if (mUnderFileSystemMap.putIfAbsent(key, fs) != null) {
+        // This shouldn't occur, warn if it
```
Comment (gpang) [inline core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java]: is this comment complete?

File: core/common/src/test/java/alluxio/util/IdUtilsTest.java [hunk 2]
```diff
@@ -55,4 +57,13 @@ public void createRpcId() throws Exception {
     assertTrue(!second.isEmpty());
     assertNotEquals(first, second);
   }
+
+  @Test
+  public void testPositiveLong() {
```
Comment (gpang) [inline core/common/src/test/java/alluxio/util/IdUtilsTest.java]: why was this test added?

File: shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java [hunk 3]
```diff
@@ -192,7 +192,7 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
       }
 
       LOG.info("YML files for description of Property Keys were created successfully.");
-    } catch (Exception e) {
+    } catch (Exception  e) {
```
Comment (gpang) [inline shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: NIT: extra space

File: shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java [hunk 4]
```diff
@@ -114,7 +114,7 @@ public static void writeCSVFile(Collection<? extends PropertyKey> defaultKeys, S
       }
 
       LOG.info("Property Key CSV files were created successfully.");
-    } catch (Exception e) {
+    } catch (RuntimeException | IOException e) {
```
Comment (gpang) [inline shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: Undo this change for another PR?

File: core/common/src/main/java/alluxio/conf/Source.java [hunk 5]
```diff
@@ -111,10 +133,14 @@ private SitePropertySource(String filename) {
...
     public boolean equals(Object o) {
+      if (o == null) {
+        return false;
+      }
+
       if (this == o) {
         return true;
       }
-      if (!(o instanceof SitePropertySource)) {
+      if (!(this.getClass().equals(o.getClass()))) {
```
Comment (gpang) [inline core/common/src/main/java/alluxio/conf/Source.java]: is this because of the inheritance?

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 6]
```diff
@@ -199,7 +201,7 @@ public void set(PropertyKey key, Object value) {
    * @param value the value for the key
    * @param source the source of the the properties (e.g., system property, default and etc)
    */
-  public void set(PropertyKey key, Object value, Source source) {
+  public void set(@Nonnull PropertyKey key, @Nonnull Object value, @Nonnull Source source) {
```
Comment (gpang) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java]: this annotation is required?

File: core/common/src/test/java/alluxio/util/IdUtilsTest.java (hunk 2 above)
Comment (ZacBlanco) [reply core/common/src/test/java/alluxio/util/IdUtilsTest.java]: outdated comment. Test already removed

File: shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java (hunk 4 above)
Comment (ZacBlanco) [reply shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: outdatd comment, already fixed

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java (hunk 6 above)
Comment (ZacBlanco) [reply core/common/src/main/java/alluxio/conf/InstancedConfiguration.java]: it's required or else spotbugs complains about possible null derefence.

AFAIK it can't determine that `Preconditions` does the null check

File: core/common/src/main/java/alluxio/conf/Source.java (hunk 5 above)
Comment (ZacBlanco) [reply core/common/src/main/java/alluxio/conf/Source.java]: yes

File: shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java (hunk 3 above)
Comment (ZacBlanco) [reply shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: outdated, already fixed

File: core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java (hunk 1 above)
Comment (ZacBlanco) [reply core/server/common/src/main/java/alluxio/underfs/AbstractUfsManager.java]: updated

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please
